deeptime/markov/msm/tram/_tram.py

```diff
@@ -0,0 +1,708 @@
+        ---------
+        dtrajs: object
+            the first element from the data tuple.
+        bias_matrices: object
+            the second element from the data tuple.
+        ttrajs: object, optional
+            the third element from the data tuple, or None.
+        """
+        dtrajs, bias_matrices, ttrajs = data[0], data[1], data[2:]
+        if ttrajs is not None and len(ttrajs) > 0:
+            ttrajs = ttrajs[0]
+        return dtrajs, bias_matrices, ttrajs
+
+    def _validate_input(self, ttrajs, dtrajs, bias_matrices):
+        """ Check type and shape of input to ensure it can be handled by the _tram_bindings module.
+        The ttrajs, dtrajs and bias matrices should all contain the same number of trajectories (i.e. shape(0) should be
+        equal for all input arrays).
+        Trajectories can vary in length, but for each trajectory the corresponding arrays in dtrajs, ttrajs and
+        bias_matrices should be of the same length (i.e. array[i].shape(0) for any i should be equal in all input arrays).
+
+        Parameters
+        ----------
+        ttrajs: array-like, optional
+            ttrajs[i] indicates for each sample in the i-th trajectory what thermodynamic state that sample was sampled
+            at. If ttrajs is None, we assume no replica exchange was done. We then assume each trajectory corresponds to
+            a unique thermodynamic state, and assign that trajectory index to the thermodynamic state index.
+        dtrajs: array-like
+            The discrete trajectories. dtrajs[i, n] contains the Markov state index for the n-th sample in the i-th
+            trajectory.
+        bias_matrices: array-like
+            The bias energy matrices. bias_matrices[i, n, l] contains the bias energy of the n-th sample from the i'th
+            trajectory, evaluated at thermodynamic state k.
+
+        Returns
+        -------
+        ttrajs: list(ndarray(n)), int32
+            The validated ttrajs converted to a list of contiguous numpy arrays.
+        dtrajs: list(ndarray(n)), int32
+            The validated dtrajs converted to a list of contiguous numpy arrays.
+        bias_matrices: List(ndarray(n,m)), float64
+            The validated bias matrices converted to a list of contiguous numpy arrays.
+        """
+        # shape and type checks
+        assert len(dtrajs) == len(bias_matrices)
+        for d in dtrajs:
+            types.ensure_integer_array(d, ndim=1)
+        for b in bias_matrices:
+            types.ensure_floating_array(b, ndim=2)
+
+        # find dimensions
+        self.n_markov_states = max(np.max(d) for d in dtrajs) + 1
+
+        if ttrajs is None or len(ttrajs) == 0:
+            # ensure it's None. empty tuple will break the call to _tram_bindings
+            ttrajs = None
+            self.n_therm_states = len(dtrajs)
+
+        # cast types and change axis order if needed
+        dtrajs = [np.require(d, dtype=np.intc, requirements='C') for d in dtrajs]
```

Comment:
`intc` resolves to `int` which is platform dependent, but `dtrajs` are array of `int32`